src/libsyntax/ext/expand.rs

```diff
@@ -454,7 +454,7 @@ impl<'a, 'b> MacroExpander<'a, 'b> {
         };
         let path = &mac.node.path;
 
-        let ident = ident.unwrap_or(keywords::Invalid.ident());
+        let ident = ident.unwrap_or_else(|| keywords::Invalid.ident());
```

Comment:
Why does clippy recommend this replacement?
It thinks `keywords::Invalid.ident()` may be "heavy"? (It's not actually, and `unwrap_or` looked better.)